### PR TITLE
Makes the default set of status codes removable via configs

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

File: plugins/gzip/configuration.cc [hunk 1]
```diff
@@ -114,6 +114,15 @@ Configuration::add_host_configuration(HostConfiguration *hc)
   host_configurations_.push_back(hc);
 }
 
+void
+HostConfiguration::update_defaults()
+{
+  // maintain backwards compatibility/usability out of the box
+  if (compressible_status_codes_.empty()) {
+    compressible_status_codes_ = {TS_HTTP_STATUS_OK, TS_HTTP_STATUS_PARTIAL_CONTENT, TS_HTTP_STATUS_NOT_MODIFIED};
```
Comment (d2r) [inline plugins/gzip/configuration.cc]: Does this duplicate an existing definition of defaults elsewhere?

Comment (SolidWallOfCode) [reply plugins/gzip/configuration.cc]: Good point. There should be a hidden static instance of this that's the hard wired default and initialization should be copying that in to the active value.

Comment (SolidWallOfCode) [reply plugins/gzip/configuration.cc]: See `HOST_RES_DEFAULT_PREFERENCE_ORDER` in `lib/ts/ink_res_init.cc` for an example.

Comment (zwoop) [reply plugins/gzip/configuration.cc]: I don't understand. You don't like having the default set in the code like this?

Comment (SolidWallOfCode) [reply plugins/gzip/configuration.cc]: I suppose it's fine if that's the only place it shows up.

Comment (d2r) [reply plugins/gzip/configuration.cc]: > I suppose it's fine if that's the only place it shows up.

Agree. That's the only reason I asked.

Comment (SolidWallOfCode) [reply plugins/gzip/configuration.cc]: So, it's impossible to configure this with an empty set of status codes?

**Pull request merged**